File: sns.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#define INFILENAME "input.txt"
#define OUTFILENAME "result_c.txt"

// Bubble sort and linear search
int sort_n_search(int arr[], int n, int val) {
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                arr[j] = arr[j] ^ arr[j + 1];
                arr[j + 1] = arr[j] ^ arr[j + 1];
                arr[j] = arr[j] ^ arr[j + 1];
            }
        }
    }
    for (int i = 0; i < n; i++) {
        if (arr[i] == val) return i;
    }
    return -1;
}

int main() {
    // Read input from file
    FILE *IFH = fopen(INFILENAME, "r");
    if (IFH == NULL) {
        printf("Failed to open input file.\n");
        return 1;
    }
    int count, target;
    fscanf(IFH, "%d %d", &count, &target);
    int arr[count];
    for (int i = 0; i < count; i++) fscanf(IFH, "%d", &arr[i]);
    fclose(IFH);

    int index = sort_n_search(arr, count, target); // Function call

    // Write search result and sorted array to output file
    FILE *OFH = fopen(OUTFILENAME, "w");
    if (OFH == NULL) {
        printf("Failed to open output file.\n");
        return 1;
    }
    if (index != -1) fprintf(OFH, "Value %d first found at index %d.\n", target, index);
    else fprintf(OFH, "Value %d not found.\n", target);
    for (int i = 0; i < count; i++) fprintf(OFH, "%d ", arr[i]);
    fclose(OFH);
}

File: sns.py
import pyinstrument

infilename = "input.txt"
outfilename = "result_py.txt"


# Bubble sort and linear search
@pyinstrument.profile()  # Decorator for profiling the function
def sort_n_search(arr, n, val):
    for i in range(n):
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
    for i in range(n):
        if arr[i] == val:
            return i
    return None


# Read input from file
with open(infilename, "r") as file:
    lines = file.readlines()
    count, target = map(int, lines[0].split())
    nums = list(map(int, lines[1].split()))

index = sort_n_search(nums, count, target)  # Function call and profiling

# Write search result and sorted array to output file
with open(outfilename, "w") as file:
    file.write(
        f"Value {target} first found at index {index}.\n"
        if index is not None
        else f"Value {target} not found."
    )
    file.write(" ".join(map(str, nums)))
